rasa_sdk/forms.py

```diff
@@ -817,7 +817,12 @@ def slots_mapped_in_domain(self, domain: "DomainDict") -> List[Text]:
         Returns:
             Slot names mapped in the domain.
         """
-        return list(domain.get("forms", {}).get(self.form_name(), {}).keys())
+        form = domain.get("forms", {}).get(self.form_name(), {})
+        if "required_slots" in form:
+            required_slots = form.get("required_slots", {}).keys()
+        else:
+            required_slots = form.keys()
```

Comment:
can we please add a test for this behavior?

Reply:
I just added a case in `pytest parametrize` for the function `test_ask_for_next_slot` so it now tests a form with the new format (with `required_slots` keyword) and the old format (without the keyword).

Reply:
Slick!